### PR TITLE
add multi backend support for fs io

### PR DESCRIPTION
**Pull request by Young-Flash**

### CODE REVIEW

**Comment by Young-Flash** [comment]
need bleeding moonrun to run test

File: fs/fs.mbt [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+/// Reads the entire contents of a file into a string.
+///
+/// # Parameters
+/// - `path`: A `String` representing the file path.
+///
+/// # Returns
+/// A `String` containing the file contents if the file exists, otherwise raises an error.
+pub fn read_file_to_string(~path : String) -> String! {
+  if not(path_exists(~path)) {
```
Comment (lijunchen) [inline fs/fs.mbt:54]: use `guard` and `type!`? FYI

```rust
fn exist(a : String) -> Bool {
  if a == "a.txt" {
    true
  } else {
    false
  }
}

type! IOError {
  FileNotFound(String)
  DirNotFound(String)
} derive(Show)

fn read_to_string(p : String) -> String! {
  guard exist(p) else { raise IOError::FileNotFound(p) }
  "hello world"
}

fn usage_example_1() -> Unit {
  let _ = try {
    let content = read_to_string!("b.txt")
    println(content)
  } catch {
    e => println(e)
  }

}

fn usage_example_2() -> Unit {
  let content = read_to_string?("a.txt")
  println(content)
}

fn main {
  usage_example_1()
  usage_example_2()
}
```

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -60,7 +60,6 @@ jobs:
           git diff --exit-code
 
   coverage-check-bleeding:
-    if: false
```
Comment (Young-Flash) [inline .github/workflows/check.yml:63]: cc @tonyfettes

**Review by Yoorkin — APPROVED** [review]
LGTM

File: fs/fs.mbt [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+/// Reads the entire contents of a file into a string.
+///
+/// # Parameters
+/// - `path`: A `String` representing the file path.
+///
+/// # Returns
+/// A `String` containing the file contents if the file exists, otherwise raises an error.
+pub fn read_file_to_string(~path : String) -> String! {
+  guard path_exists(~path) else {
+    raise IOError::NotFound("file `\{path}` does not exist")
```
Comment (lijunchen) [inline fs/fs.mbt:58]: `raise IOError::NotFound(path)` is enough here. We can customize the `Show` trait for `IOError` to reduce redundant error messages.

**Review by lijunchen — APPROVED** [review]
LGTM

**Pull request merged**